xonsh/shell.py

```diff
@@ -161,6 +161,8 @@ def push(self, line):
 
     def reset_buffer(self):
         """Resets the line buffer."""
+        cmd = ''.join(list(filter(('\n').__ne__, self.buffer)))  
```

Comment:
The reason for filtering it to sort out elements that are a single newline. When multiline input was getting added the final item was a newline.